nengo_gui/main.py

```diff
@@ -109,8 +109,11 @@ def main():
                 try:
                     wb = webbrowser.get(args.browser)
                 except webbrowser.Error:
-                    print('Known browsers: \n  %s' % 
-                          '\n  '.join(webbrowser._browsers.keys()))
+                    try:
+                        print('Known browsers: \n  %s' % 
+                              '\n  '.join(webbrowser._tryorder))
+                    except:
```

Comment:
Bare excepts are considered to be bad practice (because they also catch things like keyboard interrupts). Usually the broadest I do is `except Exception`, but here an `except (AttributeError, TypeError)` should suffice. The `AttributeError` is raised if `_tryorder` does not exist, the `TypeError` if it is a type that does not support string joining (e.g. not a list or a list with non-strings).

Reply:
Do we know that `set(webbrowser._browsers.keys()) == set(webbrowser._tryorder)`?

Reply:
> but here an except (AttributeError, TypeError) should suffice

Hmm, good point....   those two should be enough...

> Do we know that set(webbrowser._browsers.keys()) == set(webbrowser._tryorder)?

Argh, we don't technically know that.  It's true for all the implementations I can see, but there is a provision in the webbrowser.py code that it is possible to register a webbrowser and not put it on the list (if someone calls `register` with `update_tryorder` set to something other than `1` or `-1`).  However, since that would seem to be something that someone else would have to call manually, I don't see it being possible in our case.   Some future Python version might do it, but I'm not sure how much to worry about that....

Reply:
Yeah. We could also indicate that the list we print may be incomplete, but it's probably also fine as it is right now.